src/utils/Cypher.ts

```diff
@@ -26,19 +32,79 @@ export default class Cypher {
     return this.configuration;
   }
 
-  public static encrypt(data: string): string {
+  public static async detectConfigurationKey(tenantID: string): Promise<void> {
+
+    const configCryptoKey: string = Configuration.getCryptoConfig().key;
+    const keySettings = await SettingStorage.getCryptoKeySettings(tenantID);
+    let cryptoKeySetting: CryptoKeySetting;
+
+    // Check if Crypto Key settings exists
+    if (keySettings) {
+      // Detect new config Cypher key
+      if (keySettings.cryptoKey.newKey && keySettings.cryptoKey.newKey !== configCryptoKey) {
```

Comment:
You can shorten the test like that: 
`keySettings.cryptoKey?.newKey !== configCryptoKey`

Reply:
Here I need to check if newKey is undefined because if keySetting is defined, cryptoKey (corrected to crypto) will always be defined (from the way we save the settings). And here I separate the logic if in the DB we have only oldKey (corrected to formerKey) or we have oldKey and newKey (formerKey and key). The test `keySettings.cryptoKey?.newKey !== configCryptoKey` will return true while `keySettings.cryptoKey.newKey && keySettings.cryptoKey.newKey !== configCryptoKey` will return false. I tested now in the debug console. Reading about the safe navigation operator, I don't think I can use it to test that newKey is defined. Or can I?